classes/SenseiManager.php

```diff
@@ -333,4 +380,120 @@ public static function isSensei(int $user_id, System $system): bool {
+        // 30x length (5 hrs), 12x gains: 40% of standard
+        $stat_extended_train_length = $stat_train_length * 30;
+	    // Forbidden seal trainings boost
+        $stat_long_train_length *= $player->forbidden_seal->long_training_time;
+        $stat_extended_train_length = round($stat_extended_train_length * $player->forbidden_seal->extended_training_time);
+        // Minutes * cost/min * rank
+        $standard_lesson_cost = ($stat_train_length / 60) * SenseiManager::LESSON_COST_PER_MINUTE * $player->rank_num;
+        $long_lesson_cost = ($stat_long_train_length / 60) * SenseiManager::LESSON_COST_PER_MINUTE * $player->rank_num;
+        $extended_lesson_cost = ($stat_extended_train_length / 60) * SenseiManager::LESSON_COST_PER_MINUTE * $player->rank_num;
+        $cost['short'] = $standard_lesson_cost;
+        $cost['long'] = $long_lesson_cost;
+        $cost['extended'] = $extended_lesson_cost;
+        return $cost;
+    }
+
+    public static function getLessonDurationForPlayer(User $player, System $system): array {
+        $duration = [];
+        $stat_train_length = 600;
+        // 56.25% of standard
+        $stat_long_train_length = $stat_train_length * 4;
+        // 30x length (5 hrs), 12x gains: 40% of standard
+        $stat_extended_train_length = $stat_train_length * 30;
+        // Forbidden seal trainings boost
+        $stat_long_train_length *= $player->forbidden_seal->long_training_time;
+        $stat_extended_train_length = round($stat_extended_train_length * $player->forbidden_seal->extended_training_time);
+        $duration['short'] = $stat_train_length;
+        $duration['long'] = $stat_long_train_length;
+        $duration['extended'] = $stat_extended_train_length;
+        return $duration;
+    }
+
+    public static function hasSlot(int $sensei_id, System $system): bool {
+        $slot_result = $system->db->query("SELECT `students`, `temp_students` FROM `sensei` WHERE `sensei_id` = '{$sensei_id}'");
+        $result = $system->db->fetch($slot_result);
+        $students = json_decode($result['students']);
+        $temp_students = json_decode($result['temp_students'], true);
+        $student_count = count($students) + count($temp_students);
+
+        return $student_count < 3 ? true : false;
+    }
+
+    public static function logLesson(int $sensei_id, int $temp_student_id, int $lesson_duration, array $temp_students, int $yen_gained, int $time_trained, System $system): bool {
+        $db_modified = false;
+        $temp_students[$temp_student_id] = time() + $lesson_duration;
+        $temp_students = json_encode($temp_students, JSON_FORCE_OBJECT);
+        $system->db->query("UPDATE `sensei` SET `temp_students` = '{$temp_students}', `yen_gained` = {$yen_gained}, `time_trained` = {$time_trained} WHERE `sensei_id` = '{$sensei_id}'");
+        if ($system->db->last_affected_rows > 0) {
+            $db_modified = true;
+        }
+        return $db_modified;
```

Comment:
I generally lean towards being verbose and explicit but I wonder if this provides that much benefit. If anything maybe we add a comment here, but IMO the logic is straightforward enough to read
```suggestion
        return $system->db->last_affected_rows > 0;
```